### ISSUE TITLE
`no-useless-call` rule false positive

### ISSUE THREAD

**Issue by feross** [eslint/eslint#3054]
Using `1.0.0-rc1`, the unnecessary `apply` (`no-useless-call`) rule fails on common case of using `concat` to flatten an array one level deep. Consider:

```js
var arr = [].concat.apply([], [[1], [2], [3]])
console.dir(arr)
```

With config:

```
"no-useless-call": 2
```

Giving this error:

```
t.js:1:11: unnecessary ".apply()".
```

But it works if you add a space to one of the arrays:

```js
var arr = [].concat.apply([ ], [[1], [2], [3]])
console.dir(arr)
```

This looks like a bug. These are different array instances, so the rule should never report any errors here.

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by mysticatea** [comment]
```js
var arr = [].concat.apply([], [[1], [2], [3]])
```

Its result is same as

```js
var arr = [].concat([1], [2], [3])
```

So I guess the `apply()` is useless.
`apply()` is 2x slower than the normal invocation.

> But it works if you add a space to one of the arrays

Yes, I made that comparing just source code text to keep performance.  Such as `x.y().z[0]`, we can use any expression as `this`.
But I'm not sure it's the best way.  Do you have a better idea?

**Comment by mysticatea** [comment]
OK, I got an idea.
I will fix a bug about spaces and comments.
